src/fs/inotify.rs

```diff
@@ -174,14 +176,17 @@ impl<'buf, Fd: AsFd> Reader<'buf, Fd> {
     #[allow(clippy::should_implement_trait)]
     pub fn next(&mut self) -> io::Result<InotifyEvent<'_>> {
         if self.is_buffer_empty() {
-            match read_uninit(self.fd.as_fd(), self.buf).map(|(init, _)| init.len()) {
+            todo!("FIXME: see \"Why doesn't this work?\" in examples/new_read.rs");
+            /*
+            match read(self.fd.as_fd(), self.buf).map(|(init, _)| init.len()) {
```

Comment:
`&mut *self.buf` or the GAT version of the Buffer trait fixes this, as per my other comment.